src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,85 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const showAdd =
+    title.trim() && imgUrl.trim() && imdbId.trim() && imdbUrl.trim();
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    onAdd({ title, description, imgUrl, imdbUrl, imdbId });
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+    setCount(count + 1);
```

Comment:
```suggestion
    setCount(prevCount => prevCount + 1);
```

https://dev.to/gakii/functional-state-update-in-react-42io

Reply:
Thank you for the article as I do not fully understand in which cases I should use a callback